apps/server/src/modules/files-storage/uc/files-storage.uc.ts

```diff
@@ -69,11 +70,37 @@ export class FilesStorageUC {
 	public async upload(userId: EntityId, params: FileRecordParams, req: Request): Promise<FileRecord> {
 		await this.checkPermission(userId, params.parentType, params.parentId, FileStorageAuthorizationContext.create);
 
+		await this.checkStorageLocation(userId, params.storageLocation, params.storageLocationId);
+
 		const fileRecord = await this.uploadFileWithBusboy(userId, params, req);
 
 		return fileRecord;
 	}
 
+	private async checkStorageLocation(
+		userId: EntityId,
+		storageLocation: StorageLocation,
+		storageLocationId: EntityId
+	): Promise<void> {
+		if (storageLocation === StorageLocation.INSTANCE) {
+			await this.authorizationReferenceService.checkPermissionByReferences(
+				userId,
+				AuthorizableReferenceType.Instance,
+				storageLocationId,
+				AuthorizationContextBuilder.write([Permission.INSTANCE_VIEW])
+			);
+		}
+
+		if (storageLocation === StorageLocation.SCHOOL) {
+			await this.authorizationReferenceService.checkPermissionByReferences(
+				userId,
+				AuthorizableReferenceType.School,
+				storageLocationId,
+				AuthorizationContextBuilder.write([])
+			);
+		}
```

Comment:
```suggestion
                       const authorizableReferenceType = mapType(storageLocation);
			await this.authorizationReferenceService.checkPermissionByReferences(
				userId,
				authorizableReferenceType ,
				storageLocationId,
				AuthorizationContextBuilder.write([])
			);
```

Reply:
This two if cases are in a separate private function. It dont see the usecase here to make it more complicated (to map the type and do the hacky check in the rule if instance permissions are not given).
With more cases i would agree to implement a mapping and furthermore also a mapping for the authorizationContext